src/shift-reader.js

```diff
@@ -311,7 +330,7 @@ export default class Reader extends Tokenizer {
 
       if (isLeftDelimiter(tok)) {
         if (isLeftSyntax(tok)) {
-          this.insideSyntaxTemplate.push(true);
+          this.syntaxNestingDepth++;
```

Comment:
This actually won't work. The nested arrays of booleans are needed to track syntax templates inside of interpolations:

```js
#`foo ${42 + #`blah`}`  // nonsense but lexically fine
```